### ISSUE TITLE
Showing ALL `KPH:` Entries

### ISSUE THREAD

**Issue by Major-Mike** [keepassxreboot/keepassxc-browser#2177]
## Expected Behavior

When there are `KPH:` items for a given entry, it should only show those when trying to fill in that site.

## Current Behavior

When you have `KPH:` items for an entry and trying to fill in a site, it shows **ALL** `KPH:` items from all entries that have them and not just the one you are trying to fill.

## Possible Solution

Whatever current logic is grabbing everything should be adjusted to respect the site the entry is for and currently on only.

## Steps to Reproduce (for bugs)

1. Go to a site that has `KPH:` items
2. Try to fill from the attributes list
3. Observe that ALL items are there

## Debug info

KeePassXC - 2.7.7
KeePassXC-Browser - 1.9.0.3
Operating system: Win64 Edition	Windows 11 Pro, 22H2, build 22621.3447
Browser: Microsoft Edge 124.0.2478.51 (Official build) (64-bit)

**Comment by varjolintu** [comment]
Shows them where exactly?

**Comment by Major-Mike** [comment]
In the context menu when you would select what to fill.

**Comment by varjolintu** [comment]
I cannot reproduce this. The site only retrieves entries it's allowed to, and only `KPH` attributes for those entries are shown. I need more info how you can reproduce and see this.

Create two new entries for some test site, and give both `KPH` attributes, but only allow the first entry for the site. You can still see attributes from both?

**Comment by varjolintu** [comment]
Do you mean you want to show Fill Attribute items only for the one you have already filled in the input fields? Got it.
So if none are selected, all attributes are shown in the context menu, but after filling credentials, it's filtered to the selected one?

**Comment by Major-Mike** [comment]
No, so let's say you have a site abc.com and you have three attributes called KPH: One, KPH: Two, and KPH: Three. Then you have a site called 123.com and you have two attributes called KPH: One, KPH: Two. When you go to either sites and you right click on the field that needs the value of an attribute and you go to the "attributes" submenu, you will see One, One, Two, Two, Three (all of the attributes from all the entries that have them, even though you filing on site 123.com where only One, Two should appear. Does that clarify?

**Comment by varjolintu** [comment]
No, because site 123.com does not have access to site abc.com's entry, unless you using the same entry with both. Even so, you cannot know the attribute value is the same even if the names are identical. And you cannot bind an attribute to a specific site. So this scenario is still a bit unclear to me.

**Comment by Major-Mike** [comment]
I was using the example to demonstrate but I am not sure why it is not clear. If attributes are associated to a specific entry for a specific site then it should ONLY appear on that site, period. There is absolutely no reason all the attributes for all entries should show up on the context menu, why would they?

**Comment by varjolintu** [comment]
The main thing what is unclear to me:
*Why* attributes of *all* entries are shown to you because the site only has access to an entry or entries with the specific URL for that site.

If I create two new entries with URLs 123.com and abc.com, and those have different attributes, and then access 123.com, the entry (and its attributes) for abc.com is not shown.

**Comment by Major-Mike** [comment]
I believe we are talking about the same thing. That's precisely what I was wondering and why I came here to ask why it would show all that when the expectation is that it would be limited to the entry for the site that you are on.

Now if you have 5 accounts, like say Gmails, and you have attributes for all of them and you are on a page that you need to fill, ok I can get on those how all the 5 entry attributes would show up as the "site" that is bound to them is likely the same, but this wasn't the case for the instance that brought me to you.

So what do you need to do to diagnose this, please tell me, and I can provide it. I am not a novice, so you can pretty much tell me whatever you need and I should be able to give it to you. If I still need to then I will let you know and seek input on how.

**Comment by varjolintu** [comment]
At first you should take a look at the entries Properties tab when you edit them. It lists the permissions (`KeePassXC-Browser Settings`) for the extension. Have you remembered the site permissions earlier, or do you actually get a promp where it asks for them?

Also Browser Integration's Advanced settings tab would be interesting to see. If nothing else helps.

**Comment by Major-Mike** [comment]
Sorry for the delay I was out of town for work. The first part I don't quite get but the second part here you go:

![image](https://github.com/keepassxreboot/keepassxc-browser/assets/55165479/4acdd0a2-ea94-42fb-94eb-10c8559798e3)

**Comment by varjolintu** [comment]
Just edit your (problematic) entries, and go to the Properties tab where a Plugin Data table is visible.

**Comment by varjolintu** [comment]
Let's reopen this if the problem still occures. There hasn't been updates for ages.
